src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -67,24 +74,37 @@ pub enum Notification {
 #[derive(Clone,Debug,PartialEq)]
 #[derive(Serialize, Deserialize)]
 pub struct FilesystemEvent {
-    /// Path of the file that the event is about.
-    pub path : Path,
     #[allow(missing_docs)]
+    // The serialization format requires the information to be wrapped around an "event" property.
+    pub event : FilesystemEventInfo
+}
+
+
+
+// ===========================
+// === FilesystemEventInfo ===
+// ===========================
+
+/// FilesystemEvent's kind and path of the file that the event is about.
+#[derive(Clone,Debug,PartialEq)]
+#[derive(Serialize, Deserialize)]
+#[allow(missing_docs)]
+pub struct FilesystemEventInfo {
+    pub path : Path,
     pub kind : FilesystemEventKind
 }
 
+
 /// Describes kind of filesystem event (was the file created or deleted, etc.)
 #[derive(Clone,Copy,Debug,PartialEq)]
 #[derive(Serialize, Deserialize)]
 pub enum FilesystemEventKind {
-    /// A new file under path was created.
-    Created,
-    /// Existing file under path was deleted.
-    Deleted,
+    /// A new file under path was added.
+    Added,
+    /// Existing file under path was removed.
+    Removed,
```

Comment:
I would use `allow(missing_docs)` on the whole enum, as this doc is really not necessary (it doesn't tell more than the names of the fields). Of course, leave it as is if you feel differently :)